### PR TITLE
feat: parameterise busybox image repo/tag

### PR DESCRIPTION
**Pull request by fenech**
Allow the charts to be used in environments with constraints on connectivity to container image repositories.

A hard-coded name `busybox` means that the Chart tests rely on the Kubernetes cluster being able to pull from Docker Hub. We already allow the component image to be customised, so this follows the same pattern for the busybox image.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: helm/charts/hydra/values.yaml [hunk 1]
```diff
@@ -566,3 +566,8 @@ configmap:
 test:
   # -- Provide additional labels to the test pod
   labels: {}
+
+# -- use a busybox image from another repository
+busybox:
+  repository: busybox
+  tag: 1
```
Comment (Demonsthere) [inline helm/charts/hydra/values.yaml:573]:
```suggestion
test:
  # -- Provide additional labels to the test pod
  labels: {}
  # -- use a busybox image from another repository
  image:
    repository: busybox
    tag: 1
```

Comment (Demonsthere) [reply helm/charts/hydra/values.yaml:573]: Hi there, i think this format would work better, let's keep the test image option in the test object 😉

**Review by Demonsthere — CHANGES_REQUESTED** [review]
Please also set some test override for the image in the `hacks` directory :)

File: helm/charts/hydra/values.yaml (hunk 1 above)
Comment (fenech) [reply helm/charts/hydra/values.yaml:573]: Good idea, the only exception to the rule would be in `helm/charts/oathkeeper/templates/deployment-sidecar.yaml`, where the busybox image is also used. What do you think?

Comment (Demonsthere) [reply helm/charts/hydra/values.yaml:573]: Ah, there we use the image as an initcontainer, if you can please make it configurable in the `image` section like 
```yaml
image:
  initContainer:
    repository: foo
    tag: bar
```

**Review by Demonsthere — APPROVED** [review]

**Pull request merged**